src/utils.c

```diff
@@ -145,10 +160,35 @@ get_desktop_name (void)
   if (desktop_name == NULL)
     {
       xdg_current_desktop = g_getenv ("XDG_CURRENT_DESKTOP");
-      if (xdg_current_desktop != NULL) {
-        desktop_name = g_strsplit (xdg_current_desktop, ":", 0)[0];
-      }
+      if (xdg_current_desktop != NULL)
+        {
+          desktop_name = g_strsplit (xdg_current_desktop, ":", 0)[0];
+        }
     }
 
   return desktop_name;
 }
+
+const char*
+get_desktop_session (void)
+{
+  static char * desktop_session = NULL;
```

Comment:
This will need to be 
`static const char * desktop_session = NULL;`
otherwise we fail strict checks.

Reply:
> This will need to be `static const char * desktop_session = NULL;` otherwise we fail strict checks.

ACK. Amended now.